src/components/network-map-tab.js

```diff
@@ -194,78 +181,178 @@ export const NetworkMapTab = ({
+                }
+            });
+            return notFoundEquipmentsIds;
+        },
+        []
+    );
+
+    const getMissingEquipmentsPositions = useCallback(
+        (notFoundEquipmentsIds, fetchEquipmentCB) => {
+            if (notFoundEquipmentsIds.length === 0) {
+                return Promise.resolve([]);
+            }
+
+            return fetchEquipmentCB(
+                studyUuid,
+                currentNode?.id,
+                notFoundEquipmentsIds
+            );
+        },
+        [studyUuid, currentNode?.id]
+    );
+
     const reloadMapGeoData = useCallback(() => {
-        console.info(`Loading geo data of study '${studyUuid}'...`);
-        const substationPositions = fetchSubstationPositions(
-            studyUuid,
-            currentNode?.id
-        );
-        const linePositions = lineFullPath
-            ? fetchLinePositions(studyUuid, currentNode?.id)
-            : [];
-        setWaitingLoadGeoData(true);
-
-        Promise.all([substationPositions, linePositions])
-            .then((values) => {
-                const newGeoData = new GeoData();
-                newGeoData.setSubstationPositions(values[0]);
-                newGeoData.setLinePositions(values[1]);
-                setGeoData(newGeoData);
-                setWaitingLoadGeoData(false);
-            })
-            .catch(function (error) {
-                console.error(error.message);
-                setWaitingLoadGeoData(false);
-                setErrorMessage(
-                    intlRef.current.formatMessage(
-                        { id: 'geoDataLoadingFail' },
-                        { studyUuid: studyUuid }
-                    )
+        if (studyUuid && currentNode) {
+            if (
+                geoData &&
+                (geoData.substationPositionsById.size > 0 ||
+                    geoData.linePositionsById.size > 0)
+            ) {
+                const notFoundSubstationIds = getEquipmentsNotFoundIds(
+                    geoData.substationPositionsById,
+                    network.getSubstations()
                 );
-            });
-        dispatch(resetMapReloaded());
+
+                const notFoundLinesIds = lineFullPath
+                    ? getEquipmentsNotFoundIds(
+                          geoData.linePositionsById,
+                          network.getLines()
+                      )
+                    : [];
+
+                if (
+                    notFoundSubstationIds.length > 0 ||
+                    notFoundLinesIds.length > 0
+                ) {
+                    console.info(
+                        `Loading geo data of study '${studyUuid}' of missing substations '${notFoundSubstationIds}' and missing lines '${notFoundLinesIds}'...`
+                    );
+                    setWaitingLoadGeoData(true);
+
+                    const missingSubstationPositions =
+                        notFoundSubstationIds.length > 0
```

Comment:
Is it necessary to check if notFoundSubstationIds size is > 0 since this check is present in getMissingEquipmentsPositions method ?

Reply:
That's true, done